libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookAPI.cs

```diff
@@ -37,25 +38,30 @@ public FacebookAPI(string token, string secret, string apiHost = "graph.facebook
         /// <param name="path">Path to the API endpoint, for example `/me/messages`.</param>
         /// <param name="payload">An object to be sent as parameters to the API call..</param>
         /// <param name="method">HTTP method, for example POST, GET, DELETE or PUT.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
         /// <returns>A task representing the async operation.</returns>
-        public async Task<HttpResponseMessage> CallAPIAsync(string path, FacebookMessage payload, HttpMethod method = null)
+        public async Task<HttpResponseMessage> CallAPIAsync(string path, FacebookMessage payload, HttpMethod method = null, CancellationToken cancellationToken = default)
         {
-            var proof = this.GetAppSecretProof(this.token, this.secret);
+            var proof = GetAppSecretProof(_token, _secret);
 
             if (method == null)
             {
                 method = HttpMethod.Post;
             }
 
             // send the request
-            HttpRequestMessage request = new HttpRequestMessage();
-            request.RequestUri = new Uri("https://" + this.apiHost + "/" + this.apiVersion + path + "?access_token=" + this.token + "&appsecret_proof=" + proof);
-            request.Method = method;
-            /* content type json? */
+            using (var request = new HttpRequestMessage())
+            {
+                request.RequestUri = new Uri("https://" + _apiHost + "/" + _apiVersion + path + "?access_token=" + _token + "&appsecret_proof=" + proof);
```

Comment:
Why not use string interpolation?
`new Uri($"https://{ _apiHost }/{ _apiVersion + path + }?access_token={ _token }&appsecret_proof=" + proof);`